state/executor.go

```diff
@@ -918,67 +905,17 @@ func TransactionGasCost(msg *types.Transaction, isHomestead, isIstanbul bool) (u
 	return cost, nil
 }
 
-// checkDynamicFees checks correctness of the EIP-1559 feature-related fields.
-// Basically, makes sure gas tip cap and gas fee cap are good.
```

Comment:
Why have been these checks moved?

Reply:
These checks have been moved to the tx validation function in the tx pool. That is the most suitable part for this logic since this is validation so it should be done with the rest of the validation which is placed in the tx pool. 
Here: https://github.com/0xPolygon/polygon-edge/blob/develop/txpool/txpool.go#L576

Reply:
This check should be done also at the executor level. Otherwise, if by some reason a transaction slips the checks and ends up in the transition, it should fail as well.

Reply:
There is a similar check on the executor level - `subGasLimitPrice`. 
Another point is how a transaction could come to the executor level without passing a tx pool (theoretically)? And if it is possible, we have big problems :)

Reply:
Well, a malicious validator could propose a block with the wrong values. In that case, this is the only way to check if those values are valid or not.

Reply:
Got it. Reverted it back and made more accurate.

Reply:
Btw, in this case, the transaction won't pass even without this validation. Malicious validator won't achieve its goal.